modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationFlattener.java

```diff
@@ -119,12 +120,21 @@ public Void doVisitLeafNode(String key, Serializable newVal) {
             
             return null;
         }
-        
+
+        @NotNull
+        private InnerNode currentOldInnerNode() {
```

Comment:
Why this method?

Reply:
1. To reduce code duplication
2. To avoid warnings all over the place. Idea thinks that `oldInnerNodesStack.peek()` may return `null`, so the class looks like a Christmas tree, which may mask real problems highlighted as warnings. With this method, we let Idea know that it cannot return `null`.

Reply:
Let's use `Queue#element`

Reply:
Switched to `Deque#element()`, inlined the method